src/libcollections/btree/set.rs

```diff
@@ -137,9 +137,7 @@ pub struct Difference<'a, T: 'a> {
 #[stable(feature = "collection_debug", since = "1.17.0")]
 impl<'a, T: 'a + fmt::Debug> fmt::Debug for Difference<'a, T> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        f.debug_tuple("Difference")
-         .field(&self.clone())
-         .finish()
+        write!(f, "Difference({:?}, {:?})", self.a, self.b)
```

Comment:
It would be better to use `f.debug_tuple("Difference").field(&self.a).field(&self.b).finish()` but in this case as it's including every field, `#[derive(Debug)]` would suffice.

However, like other iterators in `std` this could use `f.debug_list().entries(self.clone()).finish()` to display a list of the items the iterator will return.